includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -196,7 +196,23 @@ public function update_payment_intent_ajax() {
 			$selected_upe_payment_type = ! empty( $_POST['wcpay_selected_upe_payment_type'] ) ? wc_clean( wp_unslash( $_POST['wcpay_selected_upe_payment_type'] ) ) : '';
 			$payment_country           = ! empty( $_POST['wcpay_payment_country'] ) ? wc_clean( wp_unslash( $_POST['wcpay_payment_country'] ) ) : null;
 
-			wp_send_json_success( $this->update_payment_intent( $payment_intent_id, $order_id, $save_payment_method, $selected_upe_payment_type, $payment_country, $fingerprint ), 200 );
+			try {
+				$updated_payment_intent = $this->update_payment_intent(
+					$payment_intent_id,
+					$order_id,
+					$save_payment_method,
+					$selected_upe_payment_type,
+					$payment_country,
+					$fingerprint
+				);
+			} catch ( Exception $e ) {
+				// If something goes on with API, forget the cached intent as we have no clues what exactly is wrong.
+				// It might be connectivity issues, intent state not allowing the update or post update period.
+				self::remove_upe_payment_intent_from_session();
```

Comment:
Keeping this for now. If the order changes status when an exception happens here, we can drop this statement.

Reply:
+1 for this handling.